host/xtest/pkcs11_1000.c

```diff
@@ -2565,3 +2565,277 @@ static void xtest_pkcs11_test_1011(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(pkcs11, 1011, xtest_pkcs11_test_1011,
 		 "PKCS11: Test Find Objects");
+
+static void xtest_pkcs11_test_1012(ADBG_Case_t *c)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	CK_SLOT_ID slot = 0;
+	CK_SESSION_HANDLE session = CK_INVALID_HANDLE;
+	CK_FLAGS session_flags = CKF_SERIAL_SESSION | CKF_RW_SESSION;
+	CK_OBJECT_HANDLE obj_hdl = CK_INVALID_HANDLE;
+	CK_OBJECT_HANDLE key_hdl = CK_INVALID_HANDLE;
+	size_t i = 0;
+
+	CK_OBJECT_CLASS obj_class = CKO_DATA;
+	CK_BBOOL obj_token = CK_FALSE;
+	CK_BBOOL obj_private = CK_FALSE;
+	uint8_t obj_value[5] = { 1, 2, 3, 4, 5 };
+	const char *obj_label = "Label";
+
+	CK_ATTRIBUTE object_template[] = {
+		{ CKA_CLASS, &obj_class, sizeof(CK_OBJECT_CLASS) },
+		{ CKA_TOKEN, &obj_token, sizeof(CK_BBOOL) },
+		{ CKA_PRIVATE, &obj_private, sizeof(CK_BBOOL) },
+		{ CKA_VALUE, obj_value, sizeof(obj_value) },
+		{ CKA_LABEL, (CK_UTF8CHAR_PTR)obj_label, strlen(obj_label) },
+	};
+
+	CK_OBJECT_CLASS secret_class = CKO_SECRET_KEY;
+	CK_BBOOL secret_token = CK_FALSE;
+	CK_BBOOL secret_private = CK_FALSE;
+	CK_KEY_TYPE secret_key_type = CKK_GENERIC_SECRET;
+	CK_ULONG secret_len = 32;
+	CK_MECHANISM_TYPE secret_allowed_mecha[] = { CKM_SHA_1, CKM_SHA224,
+						     CKM_SHA256 };
```

Comment:
`CKM_SHA*` are not really mechanism applicable to a generic secret. Prefer use `CKM_SHA*_HMAC` for consistency.

Reply:
OK